### PR TITLE
fix a bug in beam_search (moving log_softmax after logits_processor)

### PR DESCRIPTION
**Pull request by nonstopfor**
# What does this PR do?

Fix a bug in beam_search function. If we use `log_softmax` before `logits_processor`, then the processed logits may not represent logP correctly because some logits may be set to `-inf` while other logits remain unchanged (i.e., `sum(exp(logits))!=1)`. This can significantly influence the generation results in my cases.

### CODE REVIEW

**Comment by gante** [comment]
Hey @nonstopfor 👋 

Yes, your point is correct -- the normalization should happen after the logits processors. We have a flag to do it -- if you call `generate` with `renormalize_logits=True`, the last logit processor renormalizes your logits :)

This means this PR should not be needed -- let me know if it doesn't address your problem.

**Comment by nonstopfor** [comment]
Ok. Thanks.
